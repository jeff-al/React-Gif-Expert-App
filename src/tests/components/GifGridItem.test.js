import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe("Gif grid item test", () => {

    const title = "Soy un gif";
    const url = "https://Test.jpg";
    const wrapper = shallow(<GifGridItem title={title} url={url}/>);

    test('should match with snapshop', () => {

        expect(wrapper).toMatchSnapshot();

    });

    test('should have a <p></p> with title', () => {

        const p = wrapper.find("p");

        expect( p.text().trim() ).toBe( title );

    });

    test('should have a <img></img> with url', () => {

        const img = wrapper.find("img");

        expect( img.prop("src") ).toBe( url );
        expect( img.prop("alt") ).toBe( title );

    });

    test('should have animate fade in', () => {
        
        const div = wrapper.find("div");

        const classes = div.prop("className").split(" ");
        
        expect(classes).toContain("animate__animated")
        expect(classes).toContain("animate__bounce")

    })
    
})