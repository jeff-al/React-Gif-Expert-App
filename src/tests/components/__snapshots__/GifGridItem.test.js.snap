// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gif grid item test should match with snapshop 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Soy un gif"
    src="https://Test.jpg"
  />
  <p>
    Soy un gif
  </p>
</div>
`;
